src/Sentry.Unity.Editor/Native/MSBuildLocator.cs

```diff
@@ -0,0 +1,93 @@
+using System;
+using System.Diagnostics;
+using System.IO;
+using System.Linq;
+using Sentry.Extensibility;
+using UnityEditor.PackageManager;
+
+namespace Sentry.Unity.Editor.Native
+{
+    internal static class MSBuildLocator
+    {
+        public static void SetMSBuildPath(SentryEditorOptions editorOptions, IDiagnosticLogger? logger)
+        {
+            if (!File.Exists(editorOptions.VSWherePath))
+            {
+                logger?.LogDebug("Failed to find 'VSWhere' at '{0}'. Trying to locate.", editorOptions.VSWherePath);
+                SetVSWherePath(editorOptions, logger);
+            }
+
+            logger?.LogDebug("Using 'VSWhere' at '{0}' to locate MSBuild.", editorOptions.VSWherePath);
+
+            var vsWhereOutput = "";
+            var process = new Process
+            {
+                StartInfo = new ProcessStartInfo
+                {
+                    FileName = editorOptions.VSWherePath,
+                    Arguments = "-latest -requires Microsoft.Component.MSBuild -find MSBuild\\**\\Bin\\MSBuild.exe",
+                    UseShellExecute = false,
+                    RedirectStandardOutput = true,
+                    RedirectStandardError = true,
+                    CreateNoWindow = true
+                }
+            };
+
+            process.OutputDataReceived += (sender, args) => vsWhereOutput += args.Data;
+            process.Start();
+            process.BeginOutputReadLine();
```

Comment:
I haven't used this in years but I wonder: Can we start listening before we start the program? I wonder if we could miss out on its output due to a race here